RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducerGPU.cc

```diff
@@ -177,15 +198,15 @@ void EcalRecHitProducerGPU::acquire(edm::Event const& event,
   // - adt2gev
 
   //
-  setup.get<EcalADCToGeVConstantRcd>().get(ADCToGeVConstantHandle_);
-  setup.get<EcalIntercalibConstantsRcd>().get(IntercalibConstantsHandle_);
-  setup.get<EcalChannelStatusRcd>().get(ChannelStatusHandle_);
-
-  setup.get<EcalLaserAPDPNRatiosRcd>().get(LaserAPDPNRatiosHandle_);
-  setup.get<EcalLaserAPDPNRatiosRefRcd>().get(LaserAPDPNRatiosRefHandle_);
-  setup.get<EcalLaserAlphasRcd>().get(LaserAlphasHandle_);
-  setup.get<EcalLinearCorrectionsRcd>().get(LinearCorrectionsHandle_);
-  setup.get<JobConfigurationGPURecord>().get(recHitParametersHandle_);
+  ADCToGeVConstantHandle_ = setup.getHandle(tokenADCToGeVConstant_);
+  IntercalibConstantsHandle_ = setup.getHandle(tokenIntercalibConstants_);
+  ChannelStatusHandle_ = setup.getHandle(tokenChannelStatus_);
+
+  LaserAPDPNRatiosHandle_ = setup.getHandle(tokenLaserAPDPNRatios_);
+  LaserAPDPNRatiosRefHandle_ = setup.getHandle(tokenLaserAPDPNRatiosRef_);
+  LaserAlphasHandle_ = setup.getHandle(tokenLaserAlphas_);
+  LinearCorrectionsHandle_ = setup.getHandle(tokenLinearCorrections_);
+  recHitParametersHandle_ = setup.getHandle(tokenRecHitParameters_);
 
   auto const& ADCToGeVConstantProduct = ADCToGeVConstantHandle_->getProduct(ctx.stream());
```

Comment:
are the handles actually needed? It seems like their usefulness is only as a temporary to be immediately dereferenced.
`getData` here seems more appropriate.
```c++
   auto const& adcToGeVConstantProduct = setup.getData(tokenADCToGeVConstant_).getProduct(ctx.stream());
```

please check and cleanup.

Reply:
Only two of those are needed. I will replace the rest with `getData()`.

Reply:
> Only two of those are needed. I will replace the rest with `getData()`.

thank you.

the remaining two are used only locally and could as well be defined just in the `::acquire` method itself.